src/main/java/com/limechain/trie/BlockTrieAccessor.java

```diff
@@ -9,12 +9,17 @@
  */
 public class BlockTrieAccessor extends TrieAccessor {
 
-    public BlockTrieAccessor(Hash256 blockHash) {
-        super(null);
+    /**
+     * Constructs a new BlockTrieAccessor using a {@link Hash256} object representing the state root hash.
+     *
+     * @param stateRootHash the state root hash of the block whose trie is to be accessed
+     */
+    public BlockTrieAccessor(Hash256 stateRootHash) {
+        super(stateRootHash.getBytes());
 
         BlockState blockState = BlockState.getInstance();
-        if (blockState.isInitialized() && blockState.hasHeader(blockHash)) {
-            super.lastRoot = blockState.getHeader(blockHash).getStateRoot().getBytes();
+        if (blockState.isInitialized() && blockState.hasHeader(stateRootHash)) {
+            super.lastRoot = blockState.getHeader(stateRootHash).getStateRoot().getBytes();
```

Comment:
I think either the argument is misnamed (should remain block hash) or `blockState.getHeader` does something different from what's described (it should expect the block hash).

Reply:
Indeed it is supposed to be blockHash, the confusion came from the superclass' constructor.